src/WebJobs.Script/Diagnostics/FunctionTraceWriterFactory.cs

```diff
@@ -22,12 +24,17 @@ public TraceWriter Create()
         {
             if (_scriptHostConfig.FileLoggingMode != FileLoggingMode.Never)
             {
-                TraceLevel functionTraceLevel = _scriptHostConfig.HostConfig.Tracing.ConsoleLevel;
-                string logFilePath = Path.Combine(_scriptHostConfig.RootLogPath, "Function", _functionName);
-                return new FileTraceWriter(logFilePath, functionTraceLevel);
+                return _writerCache.GetOrAdd(_functionName, (n) => CreateFileTraceWriter());
```

Comment:
You were right. We were returning cached, disposed, FileTraceWriters. I've set it up to remove upon disposal and added an e2e test to validate we're not re-using them after restarts.